### PR TITLE
Add slack integration #67

### PR DESCRIPTION
**Pull request by bewiwi**
Add slack integration #67 

Allow the possibility to manage slack integration.

Example:

```hcl
# Create a new Datadog - Slack integration
resource "datadog_integration_slack" "foo1" {
	account = "slack_pro_account"
	url = "http:///myslack/url"
   
   channels {
	   channel_name = "#prod-outage"
	   transfer_all_user_comments = false       
   }

   channels {
	   channel_name = "#staging-outage"
	   transfer_all_user_comments = false       
   }
}
```

### CODE REVIEW

File: datadog/resource_datadog_integration_slack.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	return nil
+}
+
+func resourceDatadogIntegrationSlackRead(d *schema.ResourceData, meta interface{}) error {
+	integrationSlackMutex.Lock()
+	defer integrationSlackMutex.Unlock()
+
+	client := meta.(*datadog.Client)
+	account := d.Id()
+	d.SetId("")
```
Comment (jirikuncar) [inline datadog/resource_datadog_integration_slack.go]: why do you set it to `""` if [later](https://github.com/terraform-providers/terraform-provider-datadog/pull/411/files#diff-e35c3d1575be0cb0b644b1b7caaac061R146) you check for its value?

Comment (bewiwi) [reply datadog/resource_datadog_integration_slack.go]: I set it to blank in a first place and re-set it if hook was found. I updated code to be more readable using a boolean var instead.

**Comment by msuterski** [comment]
Thanks for adding this! We're about to move some of the manual maintenance to TF and this will help a lot.

What's the current status of this PR? Are there any plans to merge and release it any time soon?

**Comment by bkabrda** [comment]
Hey 👋 thanks a lot for sending the PR! What I'm concerned about here is a concern mentioned in [1] that we eventually had to fix for Pagerduty integration. In short, for Pagerduty integration there was no way to configure individual service objects (analogous to `channels` in the Slack integration) independently (from multiple TF configs spread across several repos). The path forward implemented in [2] isn't completely straightforward and I'd like to avoid it.

That said, the DD team responsible for this API currently has no plans to implement manipulating individual channels, so I guess we can go forward with the approach in this PR, while understanding the caveat mentioned above and bearing in mind that in future we might need to do the same thing we did for the Pagerduty integration. I'll do an in-depth review and hopefully we would get this merged before the next release.  

[1] https://github.com/terraform-providers/terraform-provider-datadog/issues/75
[2] https://github.com/terraform-providers/terraform-provider-datadog/pull/237

**Comment by tonglil** [comment]
Any update on maybe giving this a review? :)

**Comment by therve** [comment]
The integration team is pushing a proper slack channels management API. This should make the slack resource more manageable through terraform, so that we don't reproduce the issue we had with pagerduty. We'll keep you posted.

**Comment by ken5scal** [comment]
Can't wait for the functionality! Any update since the last time?

**Comment by JordanP** [comment]
Small "up" here. Slack is soooo widely used and sooo needed :) Thanks !

**Comment by andymac4182** [comment]
@therve Thanks for that update. It looks like the new APIs are available. Any chance of this resource coming soon?

**Comment by tdewitt** [comment]
Adding another "status update" comment.  This would be a pretty great addition to our pipelines.  Right now this is one of the manual prerequisites for new services and that means it's easy to miss.

**Pull request closed without merging**